src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/ElasticSearchLogStorageFactory.java

```diff
@@ -87,15 +89,123 @@ public TaskListener nodeListener(FlowNode node) throws IOException, InterruptedE
     @Override
     public AnnotatedLargeText<Executable> overallLog(Executable build, boolean complete)
     {
-      ByteBuffer buf = new ByteBuffer();
-      return new AnnotatedLargeText<>(buf, StandardCharsets.UTF_8, true, build);
+      if(!config.isReadLogsFromElasticsearch()) {
+        AnnotatedLargeText<Executable> logFromFile = tryReadLogFile(build, complete);
+        if(logFromFile != null && logFromFile.length() > 0) {
+          return logFromFile;
+        } else {
+          return new AnnotatedLargeText<Executable>(new ByteBuffer(), StandardCharsets.UTF_8, true, build);
+        }
+      }
+
+      try {
+        AnnotatedLargeText<Executable> logFromElasticsearch = new ElasticSearchLogReader(config).overallLog(build, complete);
+        if(logFromElasticsearch.length() <= 0) {
+          AnnotatedLargeText<Executable> logFromFile = tryReadLogFile(build, complete);
+          if(logFromFile != null && logFromFile.length() > 0) return logFromFile;
+        }
+        return logFromElasticsearch;
+      }
+      catch (IOException e) {
+        AnnotatedLargeText<Executable> logFromFile = tryReadLogFile(build, complete);
+        if(logFromFile != null && logFromFile.length() > 0) {
```

Comment:
What if the logfile exists but has no content ?
We would return a brokenlogstorage which would not be correct.
Though this is unlikely to happen

Reply:
You're right. And also above I removed the length check - it does not matter which empty log we return.